crates/common/consensus/src/deneb/beacon_state.rs

```diff
@@ -1312,6 +1312,31 @@ impl BeaconState {
         }
         Ok(())
     }
+
+    pub fn process_randao(&mut self, body: BeaconBlockBody) -> anyhow::Result<()> {
+        let epoch = self.get_current_epoch();
+
+        // Verify RANDAO reveal
+        if let Some(proposer) = self
+            .validators
+            .get(self.get_beacon_proposer_index()? as usize)
+        {
+            let signing_root =
+                compute_signing_root(epoch, self.get_domain(DOMAIN_RANDAO, Some(epoch))?);
+
+            let _ =
+                eth_fast_aggregate_verify(&[&proposer.pubkey], signing_root, &body.randao_reveal);
```

Comment:
```suggestion
                ensure!(eth_fast_aggregate_verify(&[&proposer.pubkey], signing_root, &body.randao_reveal)?);
```
We should check the return value.